### PR TITLE
TypeAdapter generator

### PR DESCRIPTION
**Pull request by gonzodepedro**
Adds automatic rclcpp::TypeAdapter generation for protobuf.

**Note:** Code is ready for review, documentation will come later.
**Note:** To be merged after #26

### CODE REVIEW

File: rosidl_typesupport_protobuf_cpp/resource/idl__typeadapter_protobuf_cpp.hpp.em [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+@{
+TEMPLATE(
+    'tmpl_include_directories.em',
+    header_files=header_files,
+    system_header_files=system_header_files,
+    include_directives=include_directives
+)
+}@
+
+<<<<<<< HEAD
```
Comment (FlorianReimold) [inline rosidl_typesupport_protobuf_cpp/resource/idl__typeadapter_protobuf_cpp.hpp.em]: A wild merge conflict appears 🐛

**Review by FlorianReimold — CHANGES_REQUESTED** [review]

File: rosidl_typesupport_protobuf_cpp/resource/idl__typeadapter_protobuf_cpp.hpp.em (hunk 1 above)
Comment (gonzodepedro) [reply rosidl_typesupport_protobuf_cpp/resource/idl__typeadapter_protobuf_cpp.hpp.em]: Fixed

**Comment by FlorianReimold** [comment]
Hm, GitHub now prompts me to resolve some merge conflicts between your fork and the master branch

**Comment by tfoote** [comment]
I rebased the changes onto the recent PRs. It's passing CI now. 

The copyright tests were failing due to being overly strict. I've ticketed to restore them at https://github.com/tfoote/rosidl_typesupport_protobuf/issues/11

**Comment by gonzodepedro** [comment]
**Attention: this branch has changed and been force pushed**

**Comment by FlorianReimold** [comment]
I am currently checking back with Eclipse if there are any processes to follow, as this contribution is very big and reviewing it line by line is probably not the best approach here.

**Comment by gonzodepedro** [comment]
We will try to split this into multiple shorter and hopefully easier to review PRs.

**Comment by gonzodepedro** [comment]
We have split this PR into 3 more manageable PRs.
Closing this, in favor of #30, #31 and #32

**Pull request closed without merging**